integrations/extensions/docs/elasticsearch-install-and-setup/how_to_use_web_crawler_in_elasticsearch.md

```diff
@@ -286,19 +286,69 @@ Now you can build a custom ingest pipeline for your web crawler index on Kibana,
+    Map passage = ['text': envSplit[i++]];
     ```
-  * If you need to insert `title` to the beginning of each chunked text, use the following `passage` definition statement:
+    * If you need to include `title` and `url` as fields in each passage object, use the following `passage` definition statement:
+      ```Groovy
+      Map passage = ['text': envSplit[i++], 'title': ctx['title'], 'url': ctx['url']];
+      ```
+    * If you need to insert `title` to the beginning of each chunked text, use the following `passage` definition statement:
+      ```Groovy
+      Map passage = ['text': ctx['title'] + '. ' + envSplit[i++], 'title': ctx['title'], 'url': ctx['url']];
+      ```
+  
+
+  * Support chunking with overlapping  
+    You can use the following script in the `Source` field for the `Script` processor to support chunking with overlapping
     ```Groovy
-    Map passage = ['text': ctx['title'] + '. ' + envSplit[i++], 'title': ctx['title'], 'url': ctx['url']];
+    String[] envSplit = /((?<!M(r|s|rs)\.)(?<=\.) |(?<=\!) |(?<=\?) )/.split(ctx['body_content']);
+    ctx['passages'] = [];
+
+    StringBuilder overlappingText = new StringBuilder();
+
+    int i = 0;
+    while (i < envSplit.length) {
+        StringBuilder passageText = new StringBuilder(envSplit[i]);
+        int accumLength = envSplit[i].length();
+        ArrayList accumLengths = [];
+        accumLengths.add(accumLength);
+
+        int j = i + 1;
+        while (j < envSplit.length && passageText.length() + envSplit[j].length() < params.model_limit) {
+            passageText.append(' ').append(envSplit[j]);
+            accumLength += envSplit[j].length();
+            accumLengths.add(accumLength);
+            j++;
+        }
+
+        ctx['passages'].add(['text': overlappingText.toString() + passageText.toString()]);
+        def startLength = passageText.length() * (1 - params.overlap_percentage) + 1;
+        
+        int k = Collections.binarySearch(accumLengths, (int)startLength);
+        if (k < 0) {
+            k = -k - 1;
+        }
+        overlappingText = new StringBuilder();
+        for (int l = i + k; l < j; l++) {
+            overlappingText.append(envSplit[l]).append(' ');
+        }
+
+        i = j;
+    }
     ```
+    This script splits the `body_content` into sentences using regex and combines them into `passages`. The maximum number of characters in each paggase is controlled by the `model_limit` parameter. There is a overlapping between two adjacent passages, and it is controled by the `overlap_percentage` parameter. So, `model_limit` and `overlap_percentage` need to be configured in the `Parameters` field, for example, 
+    ```json
+    {
+      "model_limit": 2048,
+      "overlap_percentage": 0.25
+    }
+    ```
+    NOTE: Chunking with 512 tokens and 25% overlapping has been shown to perform better than other strategies, so the above `Parameters` configuration is recommeded for your chunking processor. 
```

Comment:
Do you have a source you can cite for "Chunking with 512 tokens and 25% overlapping has been shown to perform better than other strategies"?  If not, we can just leave it, but it seems like the kind of claim I would usually want to link to a source for.

Reply:
Here is a source. https://techcommunity.microsoft.com/t5/ai-azure-ai-services-blog/azure-ai-search-outperforming-vector-search-with-hybrid/ba-p/3929167.

We don't know how the 25% overlapping was implemented in their experiments. Our implementation here is not exactly 512 tokens with 25% overlapping, but more of an approximation. Do you think we should include this source here?

Reply:
I think it is a good source that makes the point you are trying to make well.  I agree the details are a little different, but this at least gives people some sense of why we think around 25% or so is good.